src/queries.cpp

```diff
@@ -107,32 +107,32 @@ void perftest(const std::string &index_filename,
         logger() << "Query type: " << t << std::endl;
         std::function<uint64_t(term_id_vec)> query_fun;
         if (t == "and") {
-            query_fun = [&](term_id_vec query) { return and_query<false>()(index, query); };
+            query_fun = [&](term_id_vec query) { return and_query<IndexType, false>(index)(query); };
```

Comment:
You don't need the lambda here, do you? `and_query(index)` is already a right type of function.